### PR TITLE
feat: implementation of test-case #002: DA Nodes sync large blocks

### LINKED ISSUE #60: testground/tests: implement TC-002
We already have all the presets done in `node_sync.go` 
The only left to do is:

- [ ] Create more composition files reflecting data 
- [ ] Adjust the bandwidths
- [ ] Better documentation tracing
- [ ] Redo the block reach to a runenv param to avoid hard-coding test ending

Ref: 

1. https://github.com/celestiaorg/test-infra/blob/main/docs/test-plans/001-Big-Blocks/test-cases/tc-002-da-sync.md
2. #1 
3. #55

### PR DESCRIPTION
**Pull request by Bidon15**
- compositions are ordered by test-case -> data-sets
- adapt manifest.toml to accept roles

Should be reviewed after #59 is merged 
Close #60

### CODE REVIEW

File: tests/common/validator.go [hunk 1]
```diff
@@ -42,7 +41,7 @@ func BuildValidator(ctx context.Context, runenv *runtime.RunEnv, initCtx *run.In
 	// Orchestrator is receiving all accounts by subscription, to then
 	// execute the `add-genesis-account` command and send back to the rest
 	// of the validators to set the initial genesis.json
-	if initCtx.GlobalSeq == 1 {
+	if seq == 1 {
```
Comment (vgonkivs) [inline tests/common/validator.go:44]: nit: I would change all magic numbers to custom types.

File: tests/node-sync/run_bridge.go [hunk 2]
```diff
@@ -139,13 +137,17 @@ func RunBridgeNode(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 		return err
 	}
 
-	runenv.RecordMessage("Finished published bridgeID Addr %d", int(initCtx.GlobalSeq))
+	runenv.RecordMessage("Finished published bridgeID Addr %d", int(initCtx.GroupSeq))
 
-	eh, err = nd.HeaderServ.GetByHeight(ctx, uint64(8))
+	eh, err = nd.HeaderServ.GetByHeight(ctx, uint64(13))
```
Comment (vgonkivs) [inline tests/node-sync/run_bridge.go:142]: why 13?

File: tests/node-sync/run_full.go [hunk 3]
```diff
@@ -80,11 +80,15 @@ func RunFullNode(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 		return err
 	}
 
-	eh, err := nd.HeaderServ.GetByHeight(ctx, uint64(6))
+	eh, err := nd.HeaderServ.GetByHeight(ctx, uint64(12))
```
Comment (vgonkivs) [inline tests/node-sync/run_full.go:83]: the same: why 12?

File: tests/common/validator.go (hunk 1 above)
Comment (Bidon15) [reply tests/common/validator.go:44]: In general, I agree that we should remove all magic numbers into something meaningful

Still, here we are just receiving back the sequence number from each of the publisher and comparing who got the first sequence, so this is self explanatory

File: tests/node-sync/run_bridge.go (hunk 2 above)
Comment (Bidon15) [reply tests/node-sync/run_bridge.go:142]: Waiting for the light/full to get to 12.
I'll change the number to a runenv value

File: tests/node-sync/run_full.go (hunk 3 above)
Comment (Bidon15) [reply tests/node-sync/run_full.go:83]: We are just syncing until 12. Still I'll change to a runenv value

**Review by vgonkivs — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,17 +1,59 @@
...
+## Repo Navigation
+
+The repository is divided into 4 main directories:
+
+1. `docs`
+2. `compositions`
+3. `tests`
+4. `testkit`
+
+The order of directories above :point_up: is how the repo should be read
```
Comment (SweeXordious) [inline README.md:40]:
```suggestion
The order of directories above :point_up: is how the repo should be read.
```

File: compositions/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Compositions
+
+Please navigate to the `manifest.toml` if you want to know more about which test cases/params are defined for the compositions
+to set during test runs
+
+## local-docker
+
+This directory contains sanity compositions that can be easily run on a local PC or any small VM(e.g. DO droplet)
+The motivation is to do quick regression check-ups if any PR arises from the stack(core/app/node etc.)
```
Comment (SweeXordious) [inline compositions/README.md:9]:
```suggestion
This directory contains sanity compositions that can be easily run on a local PC or any small VM(e.g. DO droplet). The motivation is to do quick regression check-ups if any PR arises from the stack(core/app/node etc.)
```

File: compositions/README.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Compositions
+
+Please navigate to the `manifest.toml` if you want to know more about which test cases/params are defined for the compositions
+to set during test runs
```
Comment (SweeXordious) [inline compositions/README.md:4]:
```suggestion
Please navigate to the `manifest.toml` if you want to know more about which test cases/params are defined for the compositions to set during test runs.
```

File: compositions/README.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+## local-docker
+
+This directory contains sanity compositions that can be easily run on a local PC or any small VM(e.g. DO droplet)
+The motivation is to do quick regression check-ups if any PR arises from the stack(core/app/node etc.)
+
+## cluster-k8s
+
+This directory contains compositions that are described in `docs/test-plans`. The sorting of inner directories
+are following the same pattern as the test-plan to test-case placement. Namings of directories and files follow this style:
```
Comment (SweeXordious) [inline compositions/README.md:14]:
```suggestion
This directory contains compositions that are described in `docs/test-plans`. The sorting of inner directories are following the same pattern as the test-plan to test-case placement. Namings of directories and files follow this style:
```

File: compositions/cluster-k8s/tc-002-da-sync/100-100-50-1000/320-100mib-0ms.toml [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    enable_go_build_cache = true
+    enabled = true
+    go_version = "1.18"
+  [groups.build]
+  [groups.run]
+    [groups.run.test_params]
+    latency = "0"
+    bandwidth = "100Mib"
+    block-height = "12"
+    role = "light"
```
Comment (SweeXordious) [inline compositions/cluster-k8s/tc-002-da-sync/100-100-50-1000/320-100mib-0ms.toml:107]:
```suggestion
    role = "light"

```

File: compositions/cluster-k8s/tc-002-da-sync/100-100-50-1000/320-100mib-100ms.toml [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    enable_go_build_cache = true
+    enabled = true
+    go_version = "1.18"
+  [groups.build]
+  [groups.run]
+    [groups.run.test_params]
+    latency = "100"
+    bandwidth = "100Mib"
+    block-height = "12"
+    role = "light"
```
Comment (SweeXordious) [inline compositions/cluster-k8s/tc-002-da-sync/100-100-50-1000/320-100mib-100ms.toml:107]:
```suggestion
    role = "light"

```

File: compositions/cluster-k8s/tc-002-da-sync/100-100-50-1000/320-100mib-200ms.toml [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+    enable_go_build_cache = true
+    enabled = true
+    go_version = "1.18"
+  [groups.build]
+  [groups.run]
+    [groups.run.test_params]
+    latency = "200"
+    bandwidth = "100Mib"
+    block-height = "12"
+    role = "light"
```
Comment (SweeXordious) [inline compositions/cluster-k8s/tc-002-da-sync/100-100-50-1000/320-100mib-200ms.toml:107]:
```suggestion
    role = "light"

```

File: compositions/cluster-k8s/tc-002-da-sync/40-40-20-100/256-100mib-0ms.toml [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+    enable_go_build_cache = true
+    enabled = true
+    go_version = "1.18"
+  [groups.build]
+  [groups.run]
+    [groups.run.test_params]
+    latency = "0"
+    bandwidth = "100Mib"
+    block-height = "12"
+    role = "light"
```
Comment (SweeXordious) [inline compositions/cluster-k8s/tc-002-da-sync/40-40-20-100/256-100mib-0ms.toml:107]:
```suggestion
    role = "light"

```

File: compositions/cluster-k8s/tc-002-da-sync/40-40-20-100/320-100mib-100ms.toml [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+    enable_go_build_cache = true
+    enabled = true
+    go_version = "1.18"
+  [groups.build]
+  [groups.run]
+    [groups.run.test_params]
+    latency = "100"
+    bandwidth = "100Mib"
+    block-height = "12"
+    role = "light"
```
Comment (SweeXordious) [inline compositions/cluster-k8s/tc-002-da-sync/40-40-20-100/320-100mib-100ms.toml:107]:
```suggestion
    role = "light"

```

File: compositions/cluster-k8s/tc-002-da-sync/40-40-20-100/320-100mib-200ms.toml [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+    enable_go_build_cache = true
+    enabled = true
+    go_version = "1.18"
+  [groups.build]
+  [groups.run]
+    [groups.run.test_params]
+    latency = "200"
+    bandwidth = "100Mib"
+    block-height = "12"
+    role = "light"
```
Comment (SweeXordious) [inline compositions/cluster-k8s/tc-002-da-sync/40-40-20-100/320-100mib-200ms.toml:107]:
```suggestion
    role = "light"

```

File: compositions/local-docker/002-da-sync-12.toml [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+  [groups.build_config]
+    build_base_image = "golang:1.18.3"
+    enable_go_build_cache = true
+    enabled = true
+    go_version = "1.18"
+  [groups.build]
+  [groups.run]
+    [groups.run.test_params]
+    latency = "0"
+    bandwidth = "100Mib"
```
Comment (SweeXordious) [inline compositions/local-docker/002-da-sync-12.toml:103]:
```suggestion
    bandwidth = "100Mib"

```

File: testkit/README.md [hunk 15]
```diff
@@ -0,0 +1,11 @@
+# Testing Kit
+
+This directory contains all necessary implementations that are utilized in `tests` directory
+
+Mainly:
```
Comment (SweeXordious) [inline testkit/README.md:5]:
```suggestion
This directory contains all necessary implementations that are utilized in `tests` directory, mainly:
```

File: tests/README.md [hunk 16]
```diff
@@ -0,0 +1,8 @@
+# Tests
+
+Each `*.go` follows the naming of the test-case in `docs/test-plan/test-plan-id/*`. 
+
+In order to not clutter the test-file with different participants' steps, each dedicated directory is created to 
+maintain steps of each types of participants. 
+
+The `common` directory is maintaining steps for a participant that plays the same role in every of the test-case(e.g. creating a validators set)
```
Comment (SweeXordious) [inline tests/README.md:8]:
```suggestion
The `common` directory is maintaining steps for a participant that plays the same role in every of the test-case (e.g. creating a validators set).
```

**Review by SweeXordious — APPROVED** [review]
utAck. Only some personal preferences small nits

**Review by Wondertan — APPROVED** [review]
+

**Review by Wondertan — APPROVED** [review]
+

**Pull request merged**